data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/DataService.java

```diff
@@ -42,27 +37,20 @@
 
     /**
      * Request the trusted details about a client's request from persistence (what policies to apply, user details, etc)
+     *
      * @param request the client's request for a leaf resource and their unique request token
-     * @return asynchronous what rules apply when accessing the data, returned as a {@link DataReaderRequest} to pass to the data-reader
+     * @return asynchronous what rules apply when accessing the data, returned as a {@link AuthorisedData} to pass to the data-reader
      */
-    CompletableFuture<Optional<DataReaderRequest>> authoriseRequest(final DataRequest request);
+    CompletableFuture<AuthorisedData> authoriseRequest(final DataRequest request);
```

Comment:
Good point!  The implementation for this will throw a ForbiddenException when there is no data.  ForbiddenException is not a caught exception so it does not have to be declared in the implementation of this method.  What do that mean for the interface?  It seems to me that I should not mention the exception as it may not be the case in other implementation.  That leaves me with needing a mention that it can return a null.

Reply:
`@throws ForbiddenException` in the javadoc maybe?

Reply:
I have tried unsuccessfully to search to what others have done with this.  I'll use the @throws on this and the other method, read (implementation throws a ReadException).  Seems like it could be the right way to deal with it.